trino/trino.go

```diff
@@ -1649,3 +1725,21 @@ func (s *NullSlice3Map) Scan(value interface{}) error {
 	s.Valid = true
 	return nil
 }
+
+// QueryResponse object used for callback
+type QueryResponse struct {
+	ID         string    `json:"id"`
+	QueryStats stmtStats `json:"stmtStats"`
```

Comment:
how do you make use of `QueryStats` and `Error` if those are of non-public types?

Reply:
This is sample code that we are current using in our project (mind taht name is out-of-date, since code is not updated yet)

Since query response has QueryStats we can access data inside within.
```go
type QueryCallback struct {
	target *TrinoResultWithProgress
	Ctx    *ctx.Context
}

// Function called for every request made to Query
func (qc *QueryCallback) StatsUpdated(qr trino.QueryResponse) {
	percentage := float32(math.Min(100.0, float64(qr.QueryStats.CompletedSplits*100.0)/float64(qr.QueryStats.TotalSplits)))
	if qr.QueryStats.TotalSplits == 0 || qc.target.Percentage > percentage {
		return
	}
	now := time.Now()
	qc.target.QueryID = qr.ID
	qc.target.State = qr.QueryStats.State
	qc.target.Scheduled = qr.QueryStats.Scheduled
	qc.target.Nodes = qr.QueryStats.Nodes
	qc.target.TotalSplits = qr.QueryStats.TotalSplits
	qc.target.QueuedSplits = qr.QueryStats.QueuesSplits
	qc.target.CompletedSplits = qr.QueryStats.CompletedSplits
	qc.target.ProcessedRows = qr.QueryStats.ProcessedRows
	qc.target.ProcessedBytes = qr.QueryStats.ProcessedBytes
	qc.target.Percentage = percentage
	qc.target.CheckTime = now
	if _, err := qc.Ctx.MainOrm.ID(qc.target.ParamHash).
		Omit("start_time", "end_time", "logs", "fetched_line", "fetched_path").Update(*qc.target); err != nil {
		util.LogWarn("can't update trino state at trino query history table : %+v, because %+v", *qc.target, err)
	}
}
```